### PR TITLE
[interactive_media_ads] Adds support for mid-roll ads

### PR DESCRIPTION
**Pull request by bparrishMines**
Adds `ContentProgressProvider` and adds `AdsRequest.contentProgressProvider` field.

This changes the platform interface `AdsRequest` to `PlatformAdsRequest`, so the `PlatformContentProgressProvider` can be passed to it. And the app-facing `AdsRequest` can take a `ContentProgressProver`.

Fixes https://github.com/flutter/flutter/issues/154261

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/interactive_media_ads/lib/src/android/android_ads_loader.dart [hunk 1]
```diff
@@ -79,13 +80,18 @@ base class AndroidAdsLoader extends PlatformAdsLoader {
   }
 
   @override
-  Future<void> requestAds(AdsRequest request) async {
+  Future<void> requestAds(PlatformAdsRequest request) async {
```
Comment (bparrishMines) [inline packages/interactive_media_ads/lib/src/android/android_ads_loader.dart:83]: The platform implementations haven't been exposed yet, so this shouldn't be a breaking change.

File: packages/interactive_media_ads/README.md [hunk 2]
```diff
@@ -240,20 +249,43 @@ Future<void> _requestAds(AdDisplayContainer container) {
...
-Future<void> _resumeContent() {
+Future<void> _resumeContent() async {
   setState(() {
     _shouldShowContentVideo = true;
   });
-  return _contentVideoController.play();
+
+  if (_adsManager != null) {
+    _contentProgressTimer = Timer.periodic(
+      const Duration(milliseconds: 500),
```
Comment (bparrishMines) [inline packages/interactive_media_ads/README.md:265]: @harold1208 @stuartmorgan What are your thoughts on the interval to be used here in the example? This works for me on Android and iOS, but this is a Flutter specific workaround to handle the SDK running on a separate thread.

Comment (bparrishMines) [reply packages/interactive_media_ads/README.md:265]: For context, this timer updates the SDK of the progress of the content video with this being the interval between updates.

Comment (harold1208) [reply packages/interactive_media_ads/README.md:265]: Per our offline discussion with the IMA team, a 200ms interval would be the ideal value.

**Review by harold1208 — COMMENTED** [review]
Per our offline discussion with the IMA team, a 200ms interval would be the ideal value to poll content progress.

**Review by stuartmorgan — APPROVED** [review]
LGTM once the 500->200 change is made.

**Pull request merged**